clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -899,6 +907,12 @@ public void unsubscribe() {
     @Override
     public void assign(Collection<TopicPartition> partitions) {
         acquire();
+
+        if (!this.assignment().isEmpty() && this.autoCommitEnabled)
```

Comment:
This should probably be replaced with a call to `maybeAutoCommitOffsetsAsync` as well, right?